.github/workflows/build_dot.yaml

```diff
@@ -1,54 +1,37 @@
----
 name: build-dot
 
 on:
-    push:
-        branches:
-            - main
-        paths:
-            - .github/workflows/build_dot.yaml
-            - dot/**
-            - requirements*.txt
-            - '!**.md'
-    pull_request:
-        types: [opened, synchronize, reopened, ready_for_review]
-        paths:
-            - .github/workflows/build_dot.yaml
-            - dot/**
-            - requirements*.txt
-            - '!**.md'
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - "**.md"
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+    paths-ignore:
+      - "**.md"
 
 jobs:
-    build-and-test:
-        if: github.event.pull_request.draft == false
-        runs-on: ubuntu-18.04
-        steps:
-            - name: Code Checkout
-              uses: actions/checkout@v2
+  build-and-test:
+    if: github.event.pull_request.draft == false
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Code Checkout
+        uses: actions/checkout@v3
 
-            - name: Set up Python
-              uses: actions/setup-python@v1
-              with:
-                  python-version: 3.8
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.8
+          cache: 'pip'
+          cache-dependency-path: 'requirements*.txt'
 
-            - name: Set up Cache
```

Comment:
cache is removed?

Reply:
we'll be using the pip cache feature of `actions/setup-python@v4` action.